SocialNetwork/SocialNetwork.API/Controllers/UserController.cs

```diff
@@ -1,51 +1,111 @@
 ﻿using Microsoft.AspNetCore.Mvc;
 using SocialNetwork.Core.Interfaces;
 using SocialNetwork.Core.Models;
-
+using System.Xml.Linq;
 namespace SocialNetwork.API.Controllers;
-
 [ApiController]
 [Route("users")]
 public class UserController : ControllerBase
 {
-    private readonly IUserService _userService;
+    private readonly IUserService _service;
 
-    public UserController(IUserService userService)
+    public UserController(IUserService service)
     {
-        _userService = userService;
+        _service = service;
     }
-
-    // localhost:port/api/users?take={take}&skip={skip}
+    #region Get Methods
     [HttpGet]
-    public ActionResult<IEnumerable<User>> GetUsers([FromQuery]int skip = 0, [FromQuery]int take = 20)
+    public async Task<ActionResult<IEnumerable<User>>> GetAllUsers([FromQuery] int skip = 0, [FromQuery] int take = 10)
     {
-        return Ok(_userService.GetAllUsers(skip, take));
+        try
+        {
+            return Ok( await _service.GetUsers(skip, take));
+        }
+        catch(ArgumentException ex)
+        {
+            return BadRequest(ex.Message);
+        }
     }
-
+    [HttpGet("id/{id}")]
+    public async Task<ActionResult<User>> GetUserbyId([FromRoute] int id)
+    {
+        try
+        {
+            return Ok(await _service.GetUserById(id));
+        }
+        catch (ArgumentException ex)
+        {
+            return BadRequest(ex.Message);
+        }
+    }
+    [HttpGet("name/{name}")]
```

Comment:
It should look like this:
[HttpGet]
public async Task<ActionResult<User>> GetUserbyName([FromQuery] string? name = null)

it will result in an error so you should join this endpoint with the one that gets all users